src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,22 +1,75 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCondition, setTransitionConditionDescription } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
+import { useCriteriaContext } from 'components/CriteriaProvider';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
```

Comment:
I think this is a good candidate to be brought into a provider. The function is defined in `BuilderRoute` then passed through `Builder` -> `Sidebar` -> `BranchNode` -> `BranchTransition`

Reply:
Agreed, although I think this work is more in scope with [PATHWAYS-292](https://jira.mitre.org/browse/PATHWAYS-292).

Reply:
Added this comment to Pathways-292